bin/BuildPackages.sh

```diff
@@ -79,115 +135,197 @@ build_carat() {
 # It is not possible to move around compiled binaries because these have the
 # path to some data files burned in.
 zcat carat-2.1b1.tgz | tar pxf -
-ln -s carat-2.1b1/bin bin
+# If the symbolic link does not exist then create it.
+# Does not check if bin exists already with some content or as a file, etc.
+# Thus this code is unstable, but should be corrected by package author
+# and not by this script.
+if [[ ! -L ./bin ]]
+then
+  ln -s ./carat-2.1b1/bin ./bin
+fi
 cd carat-2.1b1
-make TOPDIR=`pwd`
+make TOPDIR="$(pwd)"
 chmod -R a+rX .
 cd bin
-aa=`./config.guess`
-for x in "`ls -d1 $GAPDIR/bin/${aa}*`"; do
- ln -s "$aa" "`basename $x`"
-done
+# This sets GAParch as it should be.
+# Alternatively, one could check $GAPDIR/bin for all directories instead.
+source "$GAPDIR/sysinfo.gap"
+# If the symbolic link does not exist then create it.
+if [[ ! -L ./"$GAParch" ]]
```

Comment:
Why is there an `./` there? Perhaps as a safe guard against an empty GAParch? But then we end up doing something wrong, instead of aborting, so not doing this would be preferable, or alternatively an explicit check for  GAParch being non-empty.

In short: Remove the `./`?